### PR TITLE
yum_inject_repo: handle non-root USER with install script (#177).

### PR DESCRIPTION
**Pull request by twaugh**
Inspect the base image to understand what user is being inherited.
Track changes made to that by USER instructions in the Dockerfile.
Set USER to root if needed before 'RUN rm'.
Restore USER as needed.

Please test.

### CODE REVIEW

**Comment by TomasTomecek** [comment]
Will look into this on Monday.

**Comment by twaugh** [comment]
Seems to work as intended from testing a prod run here with both (a) base layer leaving us as root and (b) base layer leaving us an non-root.

File: dock/plugins/pre_inject_yum_repo.py [hunk 1]
```diff
@@ -34,12 +34,12 @@ def add_yum_repos_to_dockerfile(yumrepos, df):
     preinsert = None
     structure = df.structure
     for insndesc in structure:
-        insn = insndesc['instruction'].lower()
-        if insn == 'maintainer':
+        insn = insndesc['instruction']
+        if insn == 'MAINTAINER':
```
Comment (goldmann) [inline dock/plugins/pre_inject_yum_repo.py]: Will it also work with lower case?

Comment (twaugh) [inline dock/plugins/pre_inject_yum_repo.py]: Yes, `insndesc['instruction']` is changed to upper-case in `Dockerfile.structure()` now.

**Comment by twaugh** [comment]
We had a mixed-case test for that before, but it got lost in the change. I've added one back in.

**Comment by goldmann** [comment]
I've tested this PR and it seems to work.

**Pull request merged**